dfget/core/downloader/p2p_downloader/p2p_downloader.go

```diff
@@ -188,13 +188,14 @@ func (p2p *P2PDownloader) Run() error {
 			} else if code == constants.CodePeerFinish {
 				p2p.finishTask(response, clientWriter)
 				return nil
-			} else if code == constants.CodePeerWait {
-				continue
-			}
-
-			logrus.Warnf("request piece result:%v", response)
-			if code == constants.CodeSourceError {
-				p2p.cfg.BackSourceReason = config.BackSourceReasonSourceError
+			} else {
```

Comment:
Redundant `else`.

Reply:
It's not redundant. It makes the `warn log` only outputted when getting an unexpected response code.

Reply:
Yes, you are right. Sorry for not have a look at the code carefully.